pyVmomi/SoapAdapter.py

```diff
@@ -1321,14 +1345,20 @@ def InvokeMethod(self, mo, info, args, outerStub=None):
       req = self.SerializeRequest(mo, info, args)
       for modifier in self.requestModifierList:
          req = modifier(req)
-      conn = self.GetConnection()
-      try:
-         conn.request('POST', self.path, req, headers)
-         resp = conn.getresponse()
-      except (socket.error, http_client.HTTPException):
-         # The server is probably sick, drop all of the cached connections.
-         self.DropConnections()
-         raise
+
+      max_retries = 100
+      num_retries = 0
+      retry = True
+      fail_next = False
+      while retry is True:
+         if num_retries == max_retries:
+            fail_next = True
+         conn, resp = self._connection(req, headers, fail_next)
+         if conn is not None:
+            retry = False
+         time.sleep(0.1)
```

Comment:
On the system I encountered the problem, the query succeeded on the second try.

The values have been selected somewhat arbitrarily, trying to balance between not hammering the server too much (thus waiting 0.1 sec), but try enough time that the server might give a proper answer within a short time.